WordPress/src/main/java/org/wordpress/android/ui/posts/PostListMainViewModel.kt

```diff
@@ -23,10 +47,24 @@ class PostListMainViewModel @Inject constructor() : ViewModel() {
     private val _isFabVisible = MutableLiveData<Boolean>()
     val isFabVisible: LiveData<Boolean> = _isFabVisible
 
+    private val _selectTab = SingleLiveEvent<Int>()
+    val selectTab = _selectTab as LiveData<Int>
+
+    private val _scrollToLocalPostId = SingleLiveEvent<Int>()
+    val scrollToLocalPostId = _scrollToLocalPostId as LiveData<Int>
+
+    private val _snackBarMessage = SingleLiveEvent<UiString>()
+    val snackBarMessage = _snackBarMessage as LiveData<UiString>
+
     fun start(site: SiteModel) {
         this.site = site
     }
 
+    override fun onCleared() {
+        super.onCleared()
+        scrollToTargetPostJob.cancel() // cancels all coroutines with the default coroutineContext
```

Comment:
I am not sure about this, but do we need to do the cleanup before `super.onCleared()`?

Reply:
Good catch :)